data/campaigns/Heir_To_The_Throne/scenarios/13_The_Dwarven_Doors.cfg

```diff
@@ -160,7 +160,78 @@
         #
         {VARIABLE_OP true_entrance_location rand "1..2"}
 
-        # TODO: manually add snow detritus
+        [lua]
+            code=<<
+
+local wml_actions = wesnoth.wml_actions
+local helper = wesnoth.require "lua/helper.lua"
+
+-- The only parameter right now is the coverage (real 0.0 to 1.0, incl.)
+-- For prime-time use the filter terrain needs to come in and the convert
+-- table needs to be VERY complete. Or, rename this to morph_terrain and
+-- have the conversion table come in as an argument, too.
+
+function wml_actions.random_snow(cfg)
+
+    local filter = {}
+    filter.terrain = 'Ce,Ch,Chr,Co,Gd,Gd^Vc,Gd^Vh,Gd^Vo,Gll^Fp,Hhd,Hhd^Fp,Hhd^Vo,Ke,Khr,Ko,Md,Mm,Rd,Re,Re^Vo,Rr,Ww'
+    local locations = wesnoth.get_locations(helper.shallow_parsed(filter))
+
+    local coverage = cfg.coverage
+    if coverage == nil or coverage < 0 then
+        coverage = 0
+    else
+        if coverage > 1 then
+            coverage = 1
+        end
+    end
+    local snowNeeded = #locations * coverage
+
+    local convert = {}
+    convert['Ce'] = 'Cea'
+    convert['Ch'] = 'Cha'
+    convert['Chr'] = 'Cha'
+    convert['Co'] = 'Coa'
+    convert['Gd'] = 'Aa'
+    convert['Gd^Vc'] = 'Aa^Vca'
+    convert['Gd^Vh'] = 'Aa^Vha'
+    convert['Gd^Vo'] = 'Aa^Voa'
+    convert['Gll^Fp'] = 'Aa^Fpa'
+    convert['Hhd'] = 'Ha'
+    convert['Hhd^Fp'] = 'Ha^Fpa'
+    convert['Hhd^Vo'] = 'Ha^Voa'
+    convert['Ke'] = 'Kea'
+    convert['Khr'] = 'Kha'
+    convert['Ko'] = 'Koa'
+    convert['Md'] = 'Ms'
+    convert['Mm'] = 'Ms'
+    convert['Rd'] = 'Aa'
+    convert['Re'] = 'Aa'
+    convert['Re^Vo'] = 'Aa^Voa'
+    convert['Rr'] = 'Aa'
+    convert['Ww'] = 'Ai'
+
+    local loopCounter
+    for loopCounter = 1, snowNeeded do
+        local locationsIndex = wesnoth.random(#locations)
+        local coordinate = locations[locationsIndex]
+        local terrainCode = wesnoth.get_terrain(coordinate[1], coordinate[2])
+        local newCode = convert[terrainCode]
+        if newCode == nil then
+            newCode = terrainCode
+        end
+        wesnoth.set_terrain(coordinate[1], coordinate[2], newCode)
+        table.remove(locations, locationsIndex)
+    end
+end
+            >>
+        [/lua]
```

Comment:
In case someone wonders/comments about it .. I first did this in WML and it took several minutes to run at 100% coverage .. in Lua it is unnoticeable whether 0% or 100% .. and then I changed the max to 33%,